src/System.Numerics.Vectors/src/System/Numerics/Vector.tt

```diff
@@ -468,49 +467,151 @@ namespace System.Numerics
                 }
-<#
-        }
-#>
                 else
                 {
                     throw new NotSupportedException(SR.Arg_TypeNotSupported);
                 }
             }
             else
             {
-<#    foreach (Type type in supportedTypes)
-    {
-#>
-                <#=GenerateIfStatementHeader(type)#>
-                {
-<#
-            for (int g = 0; g < GetNumFields(type, totalSize); g++)
-            {
-#>
-                    hash = HashHelpers.Combine(hash, this.<#=GetRegisterFieldName(type, g)#>.GetHashCode());
-<#
-            }
-#>
-                    return hash;
+                if (typeof(T) == typeof(Byte))
+                {
+                    var hashCode = new HashCode();
+                    hashCode.Add(this.register.byte_0);
+                    hashCode.Add(this.register.byte_1);
+                    hashCode.Add(this.register.byte_2);
+                    hashCode.Add(this.register.byte_3);
+                    hashCode.Add(this.register.byte_4);
+                    hashCode.Add(this.register.byte_5);
+                    hashCode.Add(this.register.byte_6);
+                    hashCode.Add(this.register.byte_7);
+                    hashCode.Add(this.register.byte_8);
+                    hashCode.Add(this.register.byte_9);
+                    hashCode.Add(this.register.byte_10);
+                    hashCode.Add(this.register.byte_11);
+                    hashCode.Add(this.register.byte_12);
+                    hashCode.Add(this.register.byte_13);
+                    hashCode.Add(this.register.byte_14);
+                    hashCode.Add(this.register.byte_15);
+                    return hashCode.ToHashCode();
+                }
+                else if (typeof(T) == typeof(SByte))
+                {
+                    var hashCode = new HashCode();
+                    hashCode.Add(this.register.sbyte_0);
+                    hashCode.Add(this.register.sbyte_1);
+                    hashCode.Add(this.register.sbyte_2);
+                    hashCode.Add(this.register.sbyte_3);
+                    hashCode.Add(this.register.sbyte_4);
+                    hashCode.Add(this.register.sbyte_5);
+                    hashCode.Add(this.register.sbyte_6);
+                    hashCode.Add(this.register.sbyte_7);
+                    hashCode.Add(this.register.sbyte_8);
+                    hashCode.Add(this.register.sbyte_9);
+                    hashCode.Add(this.register.sbyte_10);
+                    hashCode.Add(this.register.sbyte_11);
+                    hashCode.Add(this.register.sbyte_12);
+                    hashCode.Add(this.register.sbyte_13);
+                    hashCode.Add(this.register.sbyte_14);
+                    hashCode.Add(this.register.sbyte_15);
+                    return hashCode.ToHashCode();
+                }
+                else if (typeof(T) == typeof(UInt16))
+                {
+                    return HashCode.Combine(
```

Comment:
The loop is explicit because of these faster calls (HashCode.Combine is unrolled).